2023/Q3/artifacts/PoS-quint/namada-redelegation.qnt

```diff
@@ -264,11 +268,7 @@ module namada {
                                                                           .with("totalUnbonded", (curEpoch-CUBIC_OFFSET-UNBONDING_OFFSET+1).to(curEpoch+1+PIPELINE_OFFSET).mapBy(e => if (e < curEpoch+1+PIPELINE_OFFSET) validatorsState.get(v).totalUnbonded.get(e)
                                                                                                                                                                                       else Map()))
                                                                           .with("totalRedelegatedUnbonded", (curEpoch-CUBIC_OFFSET-UNBONDING_OFFSET+1).to(curEpoch+1+PIPELINE_OFFSET).mapBy(e => if (e < curEpoch+1+PIPELINE_OFFSET) validatorsState.get(v).totalRedelegatedUnbonded.get(e)
-                                                                                                                                                                                                 else Map()))                                                                                                              
-                                                                          .with("totalBonded", (curEpoch-CUBIC_OFFSET-UNBONDING_OFFSET+1).to(curEpoch+1+PIPELINE_OFFSET).mapBy(e => if (e < curEpoch+1+PIPELINE_OFFSET) validatorsState.get(v).totalBonded.get(e)
-                                                                                                                                                                                    else 0))
-                                                                          .with("totalRedelegatedBonded", (curEpoch-CUBIC_OFFSET-UNBONDING_OFFSET+1).to(curEpoch+1+PIPELINE_OFFSET).mapBy(e => if (e < curEpoch+1+PIPELINE_OFFSET) validatorsState.get(v).totalRedelegatedBonded.get(e)
-                                                                                                                                                                                               else Map()))                                                                                                                                                                                    
+                                                                                                                                                                                                 else Map()))                                                                                                                                                                                                                                                                                                 
```

Comment:
why remove these here? Seems like these lines are just clearing storage that is not going to be used again. What will the `totalBonded` and `totalRedelegatedBonded` entries still be used for possibly?

Reply:
Those lines do not make sense the way we use `totalBonded` and `totalRedelegatedBonded` in the current design. Before we were only maintaining `totalBonded` and `totalRedelegatedBonded` for an epoch window: from `curEpoch=CUBIC_OFFSET-UNBONDING_OFFSET` to `curEpoch+PIPELINE_OFFSET`, because those were the only entries required by the protocol (`slashValidator` function). Now, `totalBonded and totalRedelegatedBonded` are a strict fold over delegators of `bonded` and `redelegatedBonded` and we may need older entries in `slashValidator`.